src/map/lua/lua_zone.cpp

```diff
@@ -245,7 +245,7 @@ std::optional<CLuaBaseEntity> CLuaZone::insertDynamicEntity(sol::table table)
 
     m_pLuaZone->GetZoneEntities()->dynamicTargIds.insert(PEntity->targid);
 
-    PEntity->id = 0x1000100 + (ZoneID << 12) + PEntity->targid;
+    PEntity->id = 0x1000000 + (ZoneID << 12) + PEntity->targid;
```

Comment:
Why is the freaky little offset not needed anymore?

Reply:
I think the answer is that was a hack that was needed before #2092 to prevent crashing on windower but wasn't removed until now. Without removing that, the only thing that doesn't work is battlemod because windower's ID system actually checks targid vs ID. The client doesn't seem to care and displayed the messages anyway.

It has to do with the fact that targIDs previously started at 0x800 but were being calculated wrong when transformed to IDs.

Reply:
So, for all these entities, once we get past 0x800 (2048), we don't need that transformation? Battlemod will work for everything? What about up to 0x900 (4096)

Reply:
It should be fine, but I'll test it by spawning a ton of dynamic entities.

Reply:
Based on retail, it looks like they add 0x100 to ID if targid >= 0x800
Pic from retail, an automaton in Sih Gates (zone id 268)
![image](https://user-images.githubusercontent.com/60417494/197005951-9f7f1204-3077-44cb-8d02-33783423166b.png)
`0x1000000 + (268 << 12) + 0x819` = `17877017`
`0x1000100 + (268 << 12) + 0x819` = `17877273`

Reply:
Once I added that change in to check for 0x800 everything worked fine.

Reply:
Nice